prober/utils.go

```diff
@@ -26,6 +27,28 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var protocolToGauge = map[string]float64{
+	"ip4": 4,
+	"ip6": 6,
+}
+
+type resolver struct {
+	net.Resolver
+}
+
+// A simple wrapper around resolver.LookupIP.
+func (r *resolver) resolve(ctx context.Context, target string, protocol string) (*net.IPAddr, error) {
+	ips, err := r.LookupIP(ctx, protocol, target)
+	if err != nil {
+		return nil, err
+	}
+	for _, ip := range ips {
+		return &net.IPAddr{IP: ip}, nil
+	}
+	// Go doc did not specify when this could happen, better be defensive.
```

Comment:
The comments are clear, there's at least one address when it's nil. So this is redundant.

It's be best to inline this function, given that it's simple.

Reply:
See the other comment, the function is called twice so inlining the function will complicate the code a bit.

Reply:
Hmm, I also don't see in the doc indicating "when the err is nil, there's at least one address": https://golang.org/pkg/net/#Resolver.LookupIP

Reply:
You have to dig a bit, it's a comment on one of the underlying functions.